### ISSUE TITLE
Programmable Circuit for Coil Blocks

### ISSUE THREAD

**Issue by 0xC4DE** [GregTechCEu/GregTech#1853]
### GregTech CEu Version

N/A

### Related Problem

When CEU is used with gregicallity, you can have several fluids in an assembler multi. When one puts enough rubber in the craft a coated cable, then tries to make a coil block, the cable recipe is preferred. This can create some weird recipe conflicts. The one I'm interested in right now is the coil multiblock.

I cant imagine another scenario where this would happen in normal Gregtech, though.

### Your Solution

Add a circuit requirement for all coil blocks, say 8, or something. This will prevent recipe conflicts with cables. This might technically be a gregicallity issue, but since this mod is rarely used without it, I don't see the harm in it.

(Blame integer if this should be a gregicallity issue)

### Additional Information

I've had a couple people tell me they wouldn't mind a coil circuit recipe, I personally am not a big fan of when assembler recipes don't have a circuit, as potential recipes conflicts are more likely.

**Comment by 0xC4DE** [comment]
Please :)

**Comment by theMegaTech** [comment]
agreed, tho 8 may used for 8x wire, it'd better be something free, like 5-7 or whatever

**Comment by a-a-GiTHuB-a-a** [comment]
I think Circuit 5 would work pretty well.

**Comment by 0xC4DE** [comment]
Speaking of there has to be a well defined mathematical method to determine recipe conflicts. There exist a few small ones but they could be better detected with a simple resource matrix by circuit type

**Comment by a-a-GiTHuB-a-a** [comment]
That matrix would probably take up a decent bit of memory, though.

**Comment by 0xC4DE** [comment]
This would only exist to throw warnings at build-time

**Comment by a-a-GiTHuB-a-a** [comment]
I thought I was the only one who stored fluids in random machines. I'm too lazy to build an actual fluid tank.
Why _is_ there rubber in the machine, anyway?
